k8s/pkg/apis/longhorn/v1beta2/orphan.go

```diff
@@ -0,0 +1,70 @@
+package v1beta2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type OrphanType string
+
+const (
+	OrphanTypeReplicaDirectory = OrphanType("replica directory")
+)
+
+const (
+	OrphanConditionTypeDeletable = "Deletable"
+)
+
+const (
+	OrphanDataName = "DataName"
+	OrphanDiskFsid = "DiskFsid"
+	OrphanDiskUUID = "DiskUUID"
+	OrphanDiskPath = "DiskPath"
+)
+
+// OrphanSpec defines the desired state of the Longhorn orphaned data
+type OrphanSpec struct {
+	// The node ID on which the controller is responsible to reconcile this orphan CR.
+	// +optional
+	NodeID string `json:"nodeID"`
+	// The type of the orphaned data.
+	// Can be "replica".
+	// +optional
+	Type OrphanType `json:"orphanType"`
+
+	// The parameters of the orphaned data
+	// +optional
+	// +nullable
```

Comment:
Can this be nullable?

Reply:
No. Because the properties/parameters of the orphaned data are recorded in the map, I will remove the `// +nullable`.

Reply:
Sure. And don't forget to re-generate the CRD YAML if necessary.